### PR TITLE
Replace UnsignedMessage with fvm_shared::message::Message 

### LINKED ISSUE #1599: Replace `UnsignedMessage` with `fvm_shared::message::Message`
**Issue summary**

The `UnsignedMessage` can be replace with the similar structure from the FVM: `fvm_shared::message::Message`. Some refactoring may be required to make the switch.

https://github.com/ChainSafe/forest/blob/6bb8dde95efe0d3c05361e9f7f7d5f5c0fa1492f/vm/message/src/unsigned_message.rs#L43-L91

**Other information and links**

### PR DESCRIPTION
**Pull request by tyshko5**
**Summary of changes**
Changes introduced in this pull request:
- 

**Reference issue to close (if applicable)**

Closes #1599

**Other information and links**

### CODE REVIEW

File: blockchain/state_manager/src/lib.rs [hunk 1]
```diff
@@ -55,8 +54,7 @@ type CidPair = (Cid, Cid);
 #[derive(Serialize, Deserialize)]
 #[serde(rename_all = "PascalCase")]
 pub struct InvocResult {
-    #[serde(with = "unsigned_message::json")]
-    pub msg: UnsignedMessage,
+    pub msg: Message,
```
Comment (lemmih) [inline blockchain/state_manager/src/lib.rs:57]: Will this still be serialized correctly? I really wish we had some tests to verify the desired behavior.

File: vm/message/src/signed_message.rs [hunk 2]
```diff
@@ -66,50 +65,50 @@ impl SignedMessage {
...
-        self.message.set_gas_limit(token_amount)
+        self.message.gas_limit = token_amount;
     }
     fn set_sequence(&mut self, new_sequence: u64) {
-        self.message.set_sequence(new_sequence)
+        self.message.sequence = new_sequence;
     }
     fn required_funds(&self) -> TokenAmount {
-        self.message.required_funds()
+        self.message.gas_fee_cap.clone() * self.message.gas_limit + self.message.value.clone()
```
Comment (lemmih) [inline vm/message/src/signed_message.rs:97]:
```suggestion
        &self.message.gas_fee_cap * self.message.gas_limit + &self.message.value
```

**Review by lemmih — CHANGES_REQUESTED** [review]

File: node/rpc-api/src/data_types.rs [hunk 3]
```diff
@@ -73,8 +74,7 @@ pub type JsonRpcServerState = Arc<JsonRpcServer<JsonRpcMapRouter>>;
 // Chain API
 #[derive(Serialize, Deserialize)]
 pub struct BlockMessages {
-    #[serde(rename = "BlsMessages", with = "unsigned_message::json::vec")]
-    pub bls_msg: Vec<UnsignedMessage>,
+    pub bls_msg: Vec<Message>,
```
[the remark this answers is not in the document]
Comment (tyshko5) [reply node/rpc-api/src/data_types.rs:77]: Fixed it

File: vm/message/src/signed_message.rs (hunk 2 above)
Comment (tyshko5) [reply vm/message/src/signed_message.rs]: Fixed

File: vm/message/src/chain_message.rs [hunk 4]
```diff
@@ -1,117 +1,118 @@
...
             Self::Signed(t) => t.set_sequence(new_sequence),
-            Self::Unsigned(t) => t.set_sequence(new_sequence),
+            Self::Unsigned(t) => t.sequence = new_sequence,
         }
     }
     fn required_funds(&self) -> TokenAmount {
         match self {
             Self::Signed(t) => t.required_funds(),
-            Self::Unsigned(t) => t.required_funds(),
+            Self::Unsigned(t) => t.gas_fee_cap.clone() * t.gas_limit + t.value.clone(),
```
[the remark this answers is not in the document]
Comment (tyshko5) [reply vm/message/src/chain_message.rs]: Fixed

**Review by lemmih — APPROVED** [review]

**Review by LesnyRumcajs — APPROVED** [review]

**Pull request merged**